app/controllers/validation/resource_create_parameters.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# rubocop:disable Naming/VariableName
+
+module Validation
+  class ResourceCreateParameters
+    include ActiveModel::Validations
+
+    attr_accessor :titleName, :pubType, :packageId, :publisherName,
+                  :isPeerReviewed, :edition, :description, :url,
+                  :customCoverageList,
+                  :embargoUnit, :embargoValue, :coverageStatement
+
+    validates :titleName, presence: true
+    validates :titleName, length: { maximum: 400 }
+    validates :pubType, presence: true
+    validates :packageId, presence: true
+    validates :publisherName, length: { maximum: 250 }, allow_nil: true
+    validates :isPeerReviewed, inclusion: { in: [true, false], message: 'Invalid value' }, allow_nil: true
```

Comment:
can "isPeerreviewed" be set to nil? based on your first condition, it gives me the impression that it is a boolean and has to be one of the two true/false?